kubeaudit.go

```diff
@@ -216,6 +215,26 @@ func (r *Report) Results() []Result {
 	return results
 }
 
+// Results returns the audit results for each Kubernetes resource with a minimum severity
+func (r *Report) ResultsWithMinSeverity(minSeverity SeverityLevel) []Result {
+	var results []Result
+	for _, result := range r.results {
+		var filteredAuditResults []*AuditResult
+		for _, auditResult := range result.GetAuditResults() {
+			if auditResult.Severity >= minSeverity {
+				filteredAuditResults = append(filteredAuditResults, auditResult)
+			}
+		}
+		if len(filteredAuditResults) > 0 {
+			results = append(results, &workloadResult{
+				Resource:     result.GetResource(),
+				AuditResults: filteredAuditResults,
+			})
```

Comment:
Not sure about this part 🤔

Reply:
It looks good to me.  What is it that makes you unsure? 🤔

Reply:
I think it would be ideal to copy the original object and modify only the audit results rather than creating a new object from scratch, because now we have to specify which fields to copy (in this case only `Resource` but maybe we add more fields in the future). We can't copy the original object though because it's an interface.